bool status
int32 id